events/events.go

```diff
@@ -196,3 +204,22 @@ func (e *Events) unsubscribe(l listener) bool {
 	e.mu.Unlock()
 	return false
 }
+
+func GetContainerId(eng *engine.Engine, name string) string {
+	var buf bytes.Buffer
+	job := eng.Job("container_inspect", name)
+
+	var outStream, errStream io.Writer
+
+	outStream = &buf
+	errStream = outStream
+	job.Stdout.Set(outStream)
+	job.Stderr.Set(errStream)
+
+	if err := job.Run(); err != nil {
+		return ""
+	}
+	var out struct{ ID string }
+	json.Unmarshal(buf.Bytes(), &out)
```

Comment:
@tiborvass are you suggesting that I open the Pipe synchronously?
       r, w := io.Pipe()
      job.Stdout.Set(w)
      go func() {
           if err := job.Run(); err != nil {
           ...
     }()
If I do not do that the code gets blocked.